indices.create/20_mix_typeless_typeful yml test incorrect warning

**Build scan**:
https://gradle-enterprise.elastic.co/s/x4rshlsbvvkvq

**Repro line**:
./gradlew ':qa:smoke-test-multinode:integTestRunner' --tests "org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT" \
  -Dtests.method="test {yaml=indices.create/20_mix_typeless_typeful/Create a typeless index while there is a typed template}" \
  -Dtests.seed=1A49D0E8CAC9419D \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-SD \
  -Dtests.timezone=Europe/Moscow \
  -Druntime.java=8

**Reproduces locally?**:
Yes

**Applicable branches**:
7.x 

**Failure history**:
A small number of failures so far but it will fail every 7.x branch build

**Failure excerpt**:
```
org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT > test {yaml=indices.create/20_mix_typeless_typeful/Create a typeless index while there is a typed template} FAILED
    java.lang.AssertionError: Failure at [indices.create/20_mix_typeless_typeful:21]: got unexpected warning header [
        299 Elasticsearch-7.9.0-SNAPSHOT-72ad9a454899bf7d7ec95b370a7bbe96eff293bd "index [test-1] matches multiple legacy templates [global, test_template], composable templates will only match a single template"
    ]
```

The problem is that the expected allowed warning string does not match
```
allowed_warnings:
          - "index [test-1] matches multiple v1 templates [global, test_template], v2 index templates will only match a single index template"
``` 

In the actual message `v2 index` has changed to `composable` 
```
index [test-1] matches multiple legacy templates [global, test_template], composable templates will only match a single template
```

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/indices.create/20_mix_typeless_typeful.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/indices.create/20_mix_typeless_typeful.yml
@@ -20,7 +20,7 @@
 
  - do:
       allowed_warnings:
-          - "index [test-1] matches multiple v1 templates [global, test_template], v2 index templates will only match a single index template"
+          - "index [test-1] matches multiple legacy templates [global, test_template], composable templates will only match a single template"
       indices.create:
           index: test-1
           body:
@@ -60,8 +60,8 @@
 
  - do:
       allowed_warnings:
-          - "index [test-1] matches multiple v1 templates [global, test_template], v2 index templates will only match a single index template"
-      indices.create: 
+          - "index [test-1] matches multiple legacy templates [global, test_template], composable templates will only match a single template"
+      indices.create:
           include_type_name: true
           index: test-1
           body:
@@ -128,7 +128,7 @@
 
  - do:
       allowed_warnings:
-          - "index [test-1] matches multiple v1 templates [global, test_template], v2 index templates will only match a single index template"
+          - "index [test-1] matches multiple legacy templates [global, test_template], composable templates will only match a single template"
       index:
           index: test-1
           body: { bar: 42 }

